src/hdmf_zarr/backend.py

```diff
@@ -563,11 +579,13 @@ def __get_ref(self, ref_object):
 
         # by checking os.isdir makes sure we have a valid link path to a dir for Zarr. For conversion
         # between backends a user should always use export which takes care of creating a clean set of builders.
-        source = (builder.source
-                  if (builder.source is not None and os.path.isdir(builder.source))
-                  else self.source)
-        # Make the source relative to the current file
-        source = os.path.relpath(os.path.abspath(source), start=self.abspath)
+        # source = (builder.source
+        #           if (builder.source is not None and os.path.isdir(builder.source))
+        #           else self.source)
+        #
+        # # Make the source relative to the current file
+        # source = os.path.relpath(os.path.abspath(source), start=self.abspath)
+        source = '.'
```

Comment:
@oruebel What Ryan and I found in HDMF was that the source is always the file itself and not an external source, in this case the builder.source is the file which it was written. The equivalent was what I did which was to say `source = '.'`
Though HDF5 supports external references, we have references that are only internal. 

This plus an update to resolve_ref was our fix until we add the "queue" mechanism of writing/exporting references in the future.

Reply:
Can you clarify what you mean by "queue mechanism for export?

Reply:
@oruebel Currently in hdmf, references are resolved once everything else have been written. They are stored in a queue. 
```
@self.__queue_ref
                    def _filler():
                        self.logger.debug("Resolving object references and setting attribute on dataset '%s' "
                                          "containing attributes: %s"
                                          % (name, list(attributes.keys())))
                        refs = list()
                        for item in data:
                            refs.append(self.__get_ref(item))
                        dset = parent[name]
                        dset[()] = refs
                        self.set_attributes(dset, attributes)```
This guarantees that they will be resolved. This filler method will "fill in" on write with dset[()] = refs.

From my understanding, there is no set order to which objects are written, leaving the possibility that some references won't be properly resolved. Having them pushed to a queue to the end guarantees this resolution.

Reply:
Makes sense. Thanks for the clarification. I think this was probably added after ZarrIO was branched from HDF5IO, probably for export and that this has not made it in ZarrIO yet.